main/commons/src/main/java/org/cryptomator/common/mountpoint/MountPointChooser.java

```diff
@@ -0,0 +1,157 @@
+	 * @see #prepare(Path)
+	 */
+	Optional<Path> chooseMountPoint();
+
+	/**
+	 * Called by a {@link Volume} to prepare and/or verify the chosen Mountpoint.<br>
+	 * This method is only called if the {@link #chooseMountPoint()}-Method of the same
+	 * MountPointChooser returned a path.
+	 *
+	 * <p>Developers should override this method to prepare the Mountpoint for
+	 * the Volume and check for any obstacles that could hinder the Mount-Operation.
+	 * The Mountpoint is deemed "prepared" if it can be used to mount a Volume
+	 * without any further Filesystemactions or Userinteraction. If this is not possible,
+	 * this method should fail. In other words: This method should not return without
+	 * either failing or finalizing the preparation of the Mountpoint.
+	 * Generally speaking exceptions should be wrapped as
+	 * {@link InvalidMountPointException} to allow efficient handling by the caller.
+	 *
+	 * <p>Often the preparation of a Mountpoint involves creating files or others
+	 * actions that require cleaning up after the Volume is unmounted.
+	 * In this case developers should override the {@link #cleanup(Path)}-Method
+	 * and return {@code true} to the Volume to indicate that the
+	 * {@code #cleanup}-Method of this MPC should be called after unmount.
+	 *
+	 * <p><b>Please note:</b> If this method fails the entire
+	 * Mountpoint-Choosing-Operation is aborted without calling {@link #cleanup(Path)}
+	 * or any other MPCs. Therefore this method should do all necessary cleanup
+	 * before throwing the exception.
+	 *
+	 * @param mountPoint the Mountpoint chosen by {@link #chooseMountPoint()}
+	 * @return a boolean flag; true if cleanup is needed, false otherwise
+	 * @throws InvalidMountPointException if the preparation fails
+	 * @see #chooseMountPoint()
+	 * @see #cleanup(Path)
+	 */
+	default boolean prepare(Path mountPoint) throws InvalidMountPointException {
+		return false; //NO-OP
+	}
+
+	/**
+	 * Called by a {@link Volume} to do any cleanup needed after unmount.
+	 *
+	 * <p>This method is only called if the {@link #prepare(Path)}-Method of the same
+	 * MountPointChooser returned {@code true}. Typically developers want to
+	 * delete any files created prior to mount or do similar tasks.<br>
+	 * Exceptions in this method should be handled gracefully.
+	 *
+	 * @param mountPoint the Mountpoint that was prepared by {@link #prepare(Path)}
+	 * @see #prepare(Path)
+	 */
+	default void cleanup(Path mountPoint) {
+		//NO-OP
+	}
+
+	/**
+	 * Called by the {@link MountPointChooserModule} to sort the available MPCs
+	 * and determine their execution order.
+	 * The position must be defined by the developer to reflect a useful execution order.
+	 * MPCs with lower positions will be placed at lower indices in the resulting
+	 * {@link SortedSet} and will be executed with higher probability.
+	 * The order of execution of MPCs with equal position is undefined.
+	 *
+	 * @return the position of this MPC.
+	 */
+	int getPosition();
```

Comment:
Kind of a convention to name this "priority", if it is used to determine preference over alternatives.